### PR TITLE
Add some Media Capabilities API tests

### PR DESCRIPTION
**Pull request by chrisn**
Per spec draft at https://www.w3.org/TR/2024/WD-media-capabilities-20241007/

- Add tests for 'does not imply a codec'
- Add tests for 'valid MIME type string'
- Add tests for 'single media codec'

PTAL @markafoltz

### CODE REVIEW

File: media-capabilities/decodingInfo.any.js [hunk 1]
```diff
@@ -172,6 +186,45 @@ promise_test(t => {
...
+    type: 'file',
+    video: {
+      contentType: 'video/webm; codecs="vp09.00.10.08, vp8"',
+      width: 800,
+      height: 600,
+      bitrate: 3000,
+      framerate: '24000/1001',
+    }
+  }));
+}, "Test that decodingInfo() rejects if the video configuration contentType has a codecs parameter that indicates multiple video codecs");
```
Comment (markafoltz) [inline media-capabilities/decodingInfo.any.js:213]: For consistency with other messages:

```suggestion
}, "Test that decodingInfo rejects if the video configuration contentType has a codecs parameter that indicates multiple video codecs");
```

File: media-capabilities/decodingInfo.any.js [hunk 2]
```diff
@@ -172,6 +186,45 @@ promise_test(t => {
...
+    type: 'file',
+    video: {
+      contentType: 'video/webm; codecs="vp09.00.10.08, opus"',
+      width: 800,
+      height: 600,
+      bitrate: 3000,
+      framerate: '24000/1001',
+    }
+  }));
+}, "Test that decodingInfo() rejects if the video configuration contentType has a codecs parameter that indicates both an audio and a video codec");
```
Comment (markafoltz) [inline media-capabilities/decodingInfo.any.js:226]:
```suggestion
}, "Test that decodingInfo rejects if the video configuration contentType has a codecs parameter that indicates both an audio and a video codec");
```

File: media-capabilities/decodingInfo.any.js [hunk 3]
```diff
@@ -291,6 +352,27 @@ promise_test(t => {
...
+    audio: { contentType: 'audio/webm' },
+  }));
+}, "Test that decodingInfo rejects if the audio configuration contentType does not imply a single media codec but has no codecs parameter");
+
+promise_test(t => {
+  return promise_rejects_js(t, TypeError, navigator.mediaCapabilities.decodingInfo({
+    type: 'file',
+    audio: { contentType: 'audio/webm; codecs="vorbis, opus"' },
+  }));
+}, "Test that decodingInfo() rejects if the audio configuration contentType has a codecs parameter that indicates multiple audio codecs");
```
Comment (markafoltz) [inline media-capabilities/decodingInfo.any.js:367]:
```suggestion
}, "Test that decodingInfo rejects if the audio configuration contentType has a codecs parameter that indicates multiple audio codecs");
```

File: media-capabilities/decodingInfo.any.js [hunk 4]
```diff
@@ -291,6 +352,27 @@ promise_test(t => {
...
+
+promise_test(t => {
+  return promise_rejects_js(t, TypeError, navigator.mediaCapabilities.decodingInfo({
+    type: 'file',
+    audio: { contentType: 'audio/webm; codecs="vp09.00.10.08, opus"' },
+  }));
+}, "Test that decodingInfo() rejects if the audio configuration contentType has a codecs parameter that indicates both an audio and a video codec");
```
Comment (markafoltz) [inline media-capabilities/decodingInfo.any.js:374]:
```suggestion
}, "Test that decodingInfo rejects if the audio configuration contentType has a codecs parameter that indicates both an audio and a video codec");
```

**Review by markafoltz — CHANGES_REQUESTED** [review]
Looks good with a few minor fixups.  

Presumably we will want similar tests for `encodingInfo()`?

**Comment by markafoltz** [comment]
I requested merge rights in https://github.com/web-platform-tests/wpt/pull/49250 so I may be able to merge this myself 👏

**Comment by chrisn** [comment]
Thanks. I'll update the tests per your suggestions and add some similar tests for encodingInfo.

File: media-capabilities/decodingInfo.any.js (hunk 4 above)
Comment (chrisn) [reply media-capabilities/decodingInfo.any.js]: Fixed in https://github.com/web-platform-tests/wpt/pull/49140/commits/56583ab14e348a6daa7614c50ddac447d5ec03e9.

File: media-capabilities/decodingInfo.any.js (hunk 3 above)
Comment (chrisn) [reply media-capabilities/decodingInfo.any.js]: Fixed in https://github.com/web-platform-tests/wpt/pull/49140/commits/56583ab14e348a6daa7614c50ddac447d5ec03e9.

File: media-capabilities/decodingInfo.any.js (hunk 2 above)
Comment (chrisn) [reply media-capabilities/decodingInfo.any.js]: Fixed in https://github.com/web-platform-tests/wpt/pull/49140/commits/56583ab14e348a6daa7614c50ddac447d5ec03e9.

File: media-capabilities/decodingInfo.any.js (hunk 1 above)
Comment (chrisn) [reply media-capabilities/decodingInfo.any.js]: Fixed in https://github.com/web-platform-tests/wpt/pull/49140/commits/56583ab14e348a6daa7614c50ddac447d5ec03e9.

**Comment by chrisn** [comment]
I added encodingInfo tests, fixed a minor issue in the decodingInfo tests I added (with the framerate option), as well for your suggested edits.

File: media-capabilities/decodingInfo.any.js [hunk 5]
```diff
@@ -180,10 +233,10 @@ promise_test(t => {
       width: 800,
       height: 600,
       bitrate: 3000,
-      framerate: '24000/1001',
+      framerate: 24,
```
Comment (markafoltz) [inline media-capabilities/decodingInfo.any.js:236]: Was this change intentional?  The test suggests the framerate needs to be 'x/y'

**Review by markafoltz — APPROVED** [review]
Looks good with one small fix to look at

File: media-capabilities/decodingInfo.any.js (hunk 5 above)
Comment (chrisn) [reply media-capabilities/decodingInfo.any.js]: Sorry, no, I just fixed that.

**Review by markafoltz — APPROVED** [review]
LGTM!
